Corona-Warn-App/src/test/java/de/rki/coronawarnapp/dccreissuance/notification/DccReissuanceNotificationServiceTest.kt

```diff
@@ -71,4 +72,25 @@ internal class DccReissuanceNotificationServiceTest : BaseTest() {
             personCertificatesSettings.setDccReissuanceNotifiedAt(personIdentifier, any())
         }
     }
+
+    @Test
+    fun `dismiss the badge if the new dcc reissuance doesn't exist`() = runBlockingTest {
+        DccReissuanceNotificationService(
+            personCertificatesSettings = personCertificatesSettings,
+            personNotificationSender = personNotificationSender
+        ).notifyIfNecessary(
+            personIdentifier = personIdentifier,
+            oldWalletInfo = dccWalletInfo,
+            newWalletInfo = mockk<DccWalletInfo>().apply { every { certificateReissuance } returns null }
+        )
+
+        coEvery {
+            personCertificatesSettings.dismissReissuanceBadge(personIdentifier)
+        }
+
+        coVerify(exactly = 0) {
+            personNotificationSender.showNotification(personIdentifier, R.string.notification_body_certificate)
+            personCertificatesSettings.setDccReissuanceNotifiedAt(personIdentifier, any())
```

Comment:
Maybe change to 

```kotlin
personCertificatesSettings.setDccReissuanceNotifiedAt(any(), any())
```

so that we verify that `setDccReissuanceNotified` wasn't called no matter what the arguments are?